leftwm-core/src/handlers/command_handler.rs

```diff
@@ -143,6 +143,38 @@ fn toggle_state(state: &mut State, window_state: WindowState) -> Option<bool> {
     let window = state.focus_manager.window(&state.windows)?;
     let handle = window.handle;
     let toggle_to = !window.has_state(&window_state);
+    let tag_id = state.focus_manager.tag(0)?;
+
+    let for_active_workspace = |x: &Window| -> bool { x.tag == Some(tag_id) && x.is_managed() };
+
+    if window_state == WindowState::Fullscreen {
+        //Going to fullscreen, so we save the window order
+        //or else, we restore it!
+        if toggle_to {
+            let mut to_reorder = helpers::vec_extract(&mut state.windows, for_active_workspace);
+            let handles = to_reorder
+                .iter()
+                .map(|w| w.handle)
+                .collect::<Vec<_>>();
+
+            state.window_order = handles;
+            state.windows.append(&mut to_reorder);
```

Comment:
```suggestion
            state.window_order = state
                .windows
                .iter()
                .filter(|window| window.tag == Some(tag_id) && window.is_managed())
                .map(|w| w.handle)
                .collect::<Vec<_>>();
```
Just a suggested change that does way less moves out and into the Vec. Also includes the variable set above. 

This is stuff that happens when I am bored in the evening.